ts/src/header-validator/validate-json.ts

```diff
@@ -891,6 +906,36 @@ function channelCapacity(ctx: SourceContext, s: Source): void {
   }
 }
 
+function validateAttributionScopeFields(
+  ctx: SourceContext,
+  s: Source
+): boolean {
+  return ctx.scope('attribution_scope_limit', () => {
+    if (s.attributionScopeLimit === null) {
+      if (s.attributionScopes.size !== 0 || s.maxEventStates !== null) {
+        ctx.error(
+          'must be set if attribution_scopes or max_event_states is set'
+        )
+        return false
+      }
+      return true
+    }
+    if (
+      s.attributionScopes !== null &&
+      s.attributionScopes.size > s.attributionScopeLimit
+    ) {
+      ctx.error(
+        `the size of attribution scopes must be smaller than attribution scope limit (${s.attributionScopeLimit})`
```

Comment:
```suggestion
        `the size of attribution_scopes must be <= than attribution_scope_limit (${s.attributionScopeLimit})`
```
The condition above uses `>` so the symmetric operation is `<=`, not "smaller", which doesn't account for equality.

Reply:
Done.